### PR TITLE
Feature/main/propagate diff annotations

### PR DESCRIPTION
**Pull request by NotMyFault**

### CODE REVIEW

File: worldedit-core/src/main/java/com/fastasyncworldedit/core/command/HistorySubCommands.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package com.sk89q.worldedit.command;
+package com.fastasyncworldedit.core.command;
```
Comment (dordsor21) [inline worldedit-core/src/main/java/com/fastasyncworldedit/core/command/HistorySubCommands.java:1]: I very much disagree with having command registration classes in multiple different locations. No-one's going to be using them API-wise and it just makes it more difficult to maintain. Command stuff is hard enough to traverse as it is, and as it's just not API at all, I don't think there's much need to move it.

File: worldedit-core/src/main/java/com/fastasyncworldedit/core/command/argument/LocationConverter.java [hunk 2]
```diff
@@ -1,24 +1,7 @@
...
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <https://www.gnu.org/licenses/>.
- */
-
-package com.sk89q.worldedit.command.argument;
+package com.fastasyncworldedit.core.command.argument;
```
Comment (dordsor21) [inline worldedit-core/src/main/java/com/fastasyncworldedit/core/command/argument/LocationConverter.java:1]: Again having the command classes split across different places when they were in a much much more logical place is just going to make it harder to maintain

File: worldedit-core/src/main/java/com/fastasyncworldedit/core/internal/command/ConfirmHandler.java [hunk 3]
```diff
@@ -1,26 +1,7 @@
...
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <https://www.gnu.org/licenses/>.
- */
-
-package com.sk89q.worldedit.internal.command;
+package com.fastasyncworldedit.core.internal.command;
 
 import com.fastasyncworldedit.core.configuration.Settings;
-import com.sk89q.worldedit.command.util.annotation.Confirm;
+import com.fastasyncworldedit.core.util.annotation.Confirm;
```
Comment (dordsor21) [inline worldedit-core/src/main/java/com/fastasyncworldedit/core/internal/command/ConfirmHandler.java:4]: Same again...

File: worldedit-core/src/main/java/com/fastasyncworldedit/core/command/HistorySubCommands.java (hunk 1 above)
Comment (NotMyFault) [reply worldedit-core/src/main/java/com/fastasyncworldedit/core/command/HistorySubCommands.java:1]: Every class that is 100% added by us has been moved to Fawe equivalent packages. Annotations, command registrations or anything that has `protected`/`default` access could be moved back. A `packaged-info` for affected packages could be added for class additions.

File: worldedit-core/src/main/java/com/sk89q/worldedit/registry/state/AbstractProperty.java [hunk 4]
```diff
@@ -19,8 +19,9 @@
 
 package com.sk89q.worldedit.registry.state;
 
+import com.fastasyncworldedit.core.registry.state.PropertyKey;
 import com.fastasyncworldedit.core.util.MathMan;
-import com.sk89q.worldedit.world.block.BlockTypesCache;
+import com.sk89q.worldedit.world.block.BlockTypesCache;;
```
Comment (SirYwell) [inline worldedit-core/src/main/java/com/sk89q/worldedit/registry/state/AbstractProperty.java:24]:
```suggestion
import com.sk89q.worldedit.world.block.BlockTypesCache;
```

File: worldedit-core/src/main/java/com/sk89q/worldedit/session/request/Request.java [hunk 5]
```diff
@@ -36,28 +36,36 @@
...
     private LocalSession session;
     @Nullable
     private EditSession editSession;
+    //FAWE start
     @Nullable
     private Extent extent;
+    //FAWE end
     private boolean valid;
 
+    //FAWE start
```
Comment (SirYwell) [inline worldedit-core/src/main/java/com/sk89q/worldedit/session/request/Request.java:60]: merge FAWE stuff together

File: worldedit-core/src/main/java/com/sk89q/worldedit/world/block/BlockState.java [hunk 6]
```diff
@@ -56,24 +58,30 @@
  * An immutable class that represents the state a block can be in.
  */
 @SuppressWarnings("unchecked")
+//FAWE start - Pattern
 public class BlockState implements BlockStateHolder<BlockState>, Pattern {
```
Comment (SirYwell) [inline worldedit-core/src/main/java/com/sk89q/worldedit/world/block/BlockState.java:62]: Can be removed - BlockStateHolder already extends Pattern

File: worldedit-core/src/main/java/com/sk89q/worldedit/world/block/BlockStateHolder.java [hunk 7]
```diff
@@ -37,7 +37,9 @@
 import java.util.Map;
 import java.util.stream.Collectors;
 
+//FAWE start - TileEntityBlock, Pattern
```
Comment (SirYwell) [inline worldedit-core/src/main/java/com/sk89q/worldedit/world/block/BlockStateHolder.java:40]: Only TileEntityBlock

File: worldedit-core/src/main/java/com/sk89q/worldedit/world/registry/SimpleBlockMaterial.java [hunk 8]
```diff
@@ -42,16 +42,19 @@
     private boolean replacedDuringPlacement;
     private boolean isTranslucent;
     private boolean hasContainer;
+    //FAWE Starstart
```
Comment (SirYwell) [inline worldedit-core/src/main/java/com/sk89q/worldedit/world/registry/SimpleBlockMaterial.java:45]: typo

**Review by SirYwell — CHANGES_REQUESTED** [review]

File: worldedit-core/src/main/java/com/fastasyncworldedit/core/object/extension/platform/binding/Bindings.java [hunk 9]
```diff
@@ -1,4 +1,4 @@
-package com.sk89q.worldedit.extension.platform.binding;
+package com.fastasyncworldedit.core.object.extension.platform.binding;
```
[the remark this answers is not in the document]
Comment (NotMyFault) [reply worldedit-core/src/main/java/com/fastasyncworldedit/core/object/extension/platform/binding/Bindings.java:1]: Basically half of our classes are in the object package

I can move out named class to something more suitable but not stripping entire object rn 👀

[the remark this answers is not in the document]
Comment (SirYwell) [reply worldedit-core/src/main/java/com/fastasyncworldedit/core/object/extension/platform/binding/Bindings.java:1]: The whole parser (mask/pattern) stuff is in there too. I'd at least avoid moving new classes into that package

File: worldedit-core/src/main/java/com/fastasyncworldedit/core/command/HistorySubCommands.java (hunk 1 above)
Comment (dordsor21) [reply worldedit-core/src/main/java/com/fastasyncworldedit/core/command/HistorySubCommands.java:1]: Piston makes/requires enough of a mess as it is, I think that having FAWE command stuff is the worldedit classpath is fine as long as it's obviously so tbh

Comment (MattBDev) [reply worldedit-core/src/main/java/com/fastasyncworldedit/core/command/HistorySubCommands.java:1]: I have to agree with @dordsor21 on this. Although we should add something in maybe a  `package-info` so we know for sure what is added by FAWE.

File: worldedit-core/src/main/java/com/fastasyncworldedit/core/internal/command/ConfirmHandler.java (hunk 3 above)
Comment (MattBDev) [reply worldedit-core/src/main/java/com/fastasyncworldedit/core/internal/command/ConfirmHandler.java:4]: This one for sure will cause confusion and make it harder to traverse.

File: worldedit-core/src/main/java/com/fastasyncworldedit/core/object/extension/platform/binding/Bindings.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (MattBDev) [reply worldedit-core/src/main/java/com/fastasyncworldedit/core/object/extension/platform/binding/Bindings.java]: I've always hated the object package...

File: worldedit-core/src/main/java/com/sk89q/util/ReflectionUtil.java [hunk 10]
```diff
@@ -26,6 +26,7 @@
     private ReflectionUtil() {
     }
 
+    //FAWE start
```
[the remark this answers is not in the document]
Comment (MattBDev) [reply worldedit-core/src/main/java/com/sk89q/util/ReflectionUtil.java:29]: I'll get to work on this separately from this PR?

File: worldedit-core/src/main/java/com/sk89q/worldedit/extent/transform/BlockTransformExtent.java [hunk 11]
```diff
@@ -566,4 +577,5 @@ public final BlockState transform(BlockState block) {
     private BlockState transformInverse(BlockState block) {
         return transform(block, BLOCK_TRANSFORM_INVERSE, transformInverse);
     }
+    //FAWE ned
```
[the remark this answers is not in the document]
Comment (MattBDev) [reply worldedit-core/src/main/java/com/sk89q/worldedit/extent/transform/BlockTransformExtent.java]: Don't end Ned! He did nothing wrong. 😅

File: worldedit-core/src/main/java/com/sk89q/util/ReflectionUtil.java (hunk 10 above)
[the remark this answers is not in the document]
Comment (NotMyFault) [reply worldedit-core/src/main/java/com/sk89q/util/ReflectionUtil.java:29]: Likely, this is a quick job, but out of the scope of this PR. I focus on diffing and moving classes, not re-arranging individual methods. Feel free to raise a PR based off of this one

**Review by SirYwell — APPROVED** [review]

**Review by SirYwell — APPROVED** [review]

**Pull request merged**